src/main/python/deep_qa/training/text_trainer.py

```diff
@@ -360,34 +359,36 @@ def _get_new_embedding(self, name: str, embedding_size: int, vocab_name: str='wo
                                                name=name + '_projection')
         return embedding_layer, projection_layer
 
-    def _get_sentence_encoder(self):
+    def _get_sentence_encoder(self, encoder_type="default"):
```

Comment:
yeah, i guess i was trying to cater to the idea of having multiple instantiations of one encoder, and have "types" that are essentially templates of params that you specify that you can pick from in instantiating.

I also tried as much as possible to not conflate the "name" of the layer as keras takes it, and the "name" of the layer in the sense of which parameters to assign it. i suppose it would be better if they were one and the same, though...